test/workbox-build/node/entry-points/get-manifest.js

```diff
@@ -0,0 +1,263 @@
+const expect = require('chai').expect;
+const path = require('path');
+
+const getManifest = require('../../../../packages/workbox-build/src/entry-points/get-manifest');
+
+describe(`entry-points/get-manifest.js (End to End)`, function() {
+  const SRC_DIR = path.join(__dirname, '..', '..', 'static', 'example-project-1');
+  const BASE_OPTIONS = {
+    globDirectory: SRC_DIR,
+  };
+  const REQUIRED_PARAMS = ['globDirectory'];
+  const SUPPORTED_PARAMS = [
+    'dontCacheBustUrlsMatching',
+    'globIgnores',
+    'globPatterns',
+    'manifestTransforms',
+    'maximumFileSizeToCacheInBytes',
+    'modifyUrlPrefix',
+    'templatedUrls',
+  ].concat(REQUIRED_PARAMS);
+  const UNSUPPORTED_PARAMS = [
+    'cacheId',
+    'clientsClaim',
+    'directoryIndex',
+    'handleFetch',
+    'ignoreUrlParametersMatching',
+    'importScripts',
+    'importWorkboxFromCDN',
+    'injectionPointRegexp',
+    'navigateFallback',
+    'navigateFallbackWhitelist',
+    'runtimeCaching',
+    'skipWaiting',
+    'swSrc',
+    'swDest',
+    'swTemplate',
+  ];
+
+  describe('required parameters', function() {
+    for (const requiredParam of REQUIRED_PARAMS) {
+      it(`should reject with a ValidationError when '${requiredParam}' is missing`, async function() {
+        const options = Object.assign({}, BASE_OPTIONS);
+        delete options[requiredParam];
+
+        try {
+          await getManifest(options);
+          throw new Error('Unexpected success.');
+        } catch (error) {
+          expect(error.name).to.eql('ValidationError');
+          expect(error.details[0].context.key).to.eql(requiredParam);
+        }
+      });
+    }
+  });
+
+  describe('unsupported parameters', function() {
+    for (const unsupportedParam of UNSUPPORTED_PARAMS) {
+      it(`should reject with a ValidationError when '${unsupportedParam}' is present`, async function() {
+        const options = Object.assign({}, BASE_OPTIONS);
+        options[unsupportedParam] = unsupportedParam;
+
+        try {
+          await getManifest(options);
+          throw new Error('Unexpected success.');
+        } catch (error) {
+          expect(error.name).to.eql('ValidationError');
+          expect(error.details[0].context.key).to.eql(unsupportedParam);
+        }
+      });
+    }
+  });
+
+  describe('invalid parameter values', function() {
+    for (const param of SUPPORTED_PARAMS) {
+      it(`should reject with a ValidationError when '${param}' is null`, async function() {
+        const options = Object.assign({}, BASE_OPTIONS);
+        options[param] = null;
```

Comment:
This is not quite as long as @gauntface's original test cases, where most parameters were passed a variety of bad inputs, but I figure that `joi` validation should handle the nuance, and we can just test for a single failure.